### ISSUE TITLE
Accordian Menu for Field Properties

### ISSUE THREAD

**Issue by JoelProminic** [Moonshine-IDE/Moonshine-IDE#1031]
We have been adding collapsible sections for the new properties added in #1008.
![image](https://user-images.githubusercontent.com/24532731/169396121-0e9984d2-6c36-457f-95f0-c299e8e18558.png)

However, with the current implementation:
- It is hard to distinguish these collapsible sections from normal properties.
- It seems like the checkbox setting should be saved, but it doesn't have any practical value.

Domino Designer handles these sections with tabs.  However, @JustinProminic would like to move away from this and use an accordian container instead.  You can see an example of this in Tour de Flex:  MX Framework Components > Visual Components > Containers > Accordian.

The tabs in Designer are:
- Field Info:  Name, type, multivalue, etc.  We can use this as a catchall for other properties until we find better places for them.
- Control:  we aren't planning to convert this at the moment
- Advanced:  Help Description, Field Hint, Multi-value Options, Security Options
- Font:  Font, Size, Style, Color
- Paragraph Alignment
- Paragraph Hide When:  Let's change this to "Hide Options"
- Field Extra HTML

**Comment by feather812002** [comment]
I work around this task in today ,
Now, it look like as follow:
![image](https://user-images.githubusercontent.com/8543391/169817257-30429004-ecb5-4820-b3eb-945c79f91cee.png)
![image](https://user-images.githubusercontent.com/8543391/169817286-afe4a000-5d77-49c3-acfa-329d03876901.png)


I hope this match the require from Justin.

**Comment by JoelProminic** [comment]
The accordian panels are mostly working well, but it is just a bit too tall for the default console height.

![image](https://user-images.githubusercontent.com/24532731/179088967-50f69c8c-feb3-4095-8e8c-b8227a4b9021.png)

If I want to select the last panel,  I have to scroll down, and then the field names at the top of the panel are no longer visible.  

How is the panel height set currently?  Is this based on the height of the tallest panel (Field Info)?

Would it be possible to set the accordian panel height to fit the available vertical space?  The individual panels would need scroll bars instead, so I think it would still use roughly the same horizontal space.

**Comment by feather812002** [comment]
@JoelProminic 
I review again in today about the height on the Accordian Menu.
It default setting the height to 100% of the accordian panel
and in the panel of Accordian , the child menu ,like the "Field Info","Advanced",etcs, also is setting to 100%, 
So, the heigh  will auto fit the content of the child menu panel , so some time will have a have to scroll down to select other children menu, 
Now, it look like this:
![image](https://user-images.githubusercontent.com/8543391/185901839-a6a8904b-f4c2-4668-b816-dcadccc13420.png)

If we want short the height of the Accordian Menu , we have to add some scroll bar into the inside of children panel, 
It will let the UI become more difficult work .

We can discuss this in tonight.

**Comment by feather812002** [comment]
Hello , @JoelProminic 
I check the accofdian mean again , the source code already setting the height as "100%"
https://github.com/Moonshine-IDE/MockupVisualEditor/blob/26dca4799b9229f2bc5fb041e893605b43870ebf/src/view/domino/propertyEditors/InputTextPropertyEditor.mxml#L693

So, I don't think we miss the 100% in here , beside the height of the "Accordian Menu",
I also setting the vbox's percentHeight as "100"


https://github.com/Moonshine-IDE/MockupVisualEditor/blob/26dca4799b9229f2bc5fb041e893605b43870ebf/src/view/domino/propertyEditors/InputTextPropertyEditor.mxml#L694

**Comment by JoelProminic** [comment]
@rat-moonshine, can you take a look at this now that it has been merged to master?  I suspect there is a missing height setting on one of the parent containers

**Comment by rat-moonshine** [comment]
I looked into this today. Unfortunately, we can't do much of a thing to this as it appears, based on how the `Accordion` component build.

I did some tests, along with a simple `Accordion` component with dummy content, and it also caused the vertical-scroll to appear despite the `Accordion` height set to 100%.

![image](https://user-images.githubusercontent.com/24370870/195830613-cc44606f-5ebe-45b4-8f5c-4286034df634.png)

It tells that the sizing of the selected accordion-branch sets based on the children content height. The [API](https://flex.apache.org/asdoc/mx/containers/Accordion.html) confirms the behaviour here:

![image](https://user-images.githubusercontent.com/24370870/195830970-4c374bb1-b514-4383-9c1c-41dd1d5bede5.png)

The only adjustment I think we can takes are:

1. Set `accordion. resizeToContent = true` - this shall resize the accordion to its parent height where content is not that much
2. Set the accordion a static height rather than a percentage-height that match its parent, this will force the branch-children to clip when their height will exceed the accordion height (based on the **Child sizing rules** in above table). I confirm the scenario in following image, however setting-up the accordion with a dynamic static height could be troublesome.

![image](https://user-images.githubusercontent.com/24370870/195831956-3ddececf-bf38-4d9d-9ed9-bfb1d121d2cb.png)
